File: codes/javascript/src/function_iife.js
/* eslint-disable wrap-iife */
/* eslint-disable func-names */

console.log("// Immediately Invoked Function Expression (IIFE) in JavaScript");
console.log("// a Self-Executing Anonymous Function that runs as soon as it is defined");
console.log("// https://developer.mozilla.org/en-US/docs/Glossary/IIFE");

(function () {
    console.log("// self-executing after defined");
})();

(() => {
    console.log("// self-executing after defined");
})();

(() => console.log("// self-executing after defined"))();

File: codes/python/function_iife.py
print("# Immediately Invoked Function Expression (IIFE) in Python")
print("# a Self-Executing Anonymous Function that runs as soon as it is defined")
print("# https://developer.mozilla.org/en-US/docs/Glossary/IIFE")

@lambda _: _()
def _():
    print("# self-executing after defined")


(lambda : print("# self-executing after defined"))()
